### PR TITLE
[Hub Generated] Review request for Microsoft.RecoveryServices to add version preview/2022-09-30-preview

### PR DESCRIPTION
**Pull request by amchandn**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/amchandn/azure-rest-api-specs/release-recoveryservices-Microsoft.RecoveryServices-2022-10-02-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by amchandn** [comment]
We got the changes reviewed and approved as part of swagger PR- [[Hub Generated] Review request for Microsoft.RecoveryServices to add version preview/2022-09-01-preview by pratimaupadhyay02 · Pull Request #20578 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/20578).

However, due to timeline changes and feature requirements - we had to change the API version label from 2022-09-01-preview to 2022-10-02-preview.
Hence, I'd need to close the older pull request and requesting your sign-off on current PR -
[#20934](https://github.com/Azure/azure-rest-api-specs/pull/20934)

**Comment by xiaoxuqi-ms** [comment]
@amchandn Please fix the linter error
@tadelesh could you please help to take a look at the go breaking change

**Comment by xiaoxuqi-ms** [comment]
@amchandn understand, but we still need ARM team to sign off. Since the changes are approved, it should be easier to review in this pr.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        "localizedValue": {
+          "description": "Localized value of usage.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionId": {
+      "name": "subscriptionId",
```
Comment (jianyexi) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json:141]: please use common type instead

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      "in": "path",
+      "description": "The subscription Id.",
+      "required": true,
+      "type": "string"
+    },
+    "ResourceGroupName": {
+      "name": "resourceGroupName",
```
Comment (jianyexi) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json:148]: please use common type instead

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      "name": "vaultName",
+      "in": "path",
+      "description": "The name of the recovery services vault.",
+      "required": true,
+      "x-ms-parameter-location": "method",
+      "type": "string"
+    },
+    "ApiVersion": {
+      "name": "api-version",
+      "in": "query",
```
Comment (jianyexi) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json:165]: please use common type instead

Comment (amchandn) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-10-02-preview/vaultusages.json:165]: Yes, done

**Comment by amchandn** [comment]
We have the GET, PUT and PATCH available on "Vault" type which inherits the RecoveryServicesTrackedResource type, hence these errors are false alarms
image

**Comment by amchandn** [comment]
@jianyexi , can you please approve this PR or suggest anymore changes?

File: specification/recoveryservices/resource-manager/readme.md [hunk 4]
```diff
@@ -43,6 +43,17 @@ message-format: json
 ```
 
 
+### Tag: package-preview-2022-10
```
Comment (xiaoxuqi-ms) [inline specification/recoveryservices/resource-manager/readme.md:46]: Confirmed with @jianyexi , the linter error can be resolved by adding suppression: https://github.com/Azure/azure-rest-api-specs/pull/2229/files#diff-c52cea7cbbcf550e62037de87a237d27R50

Comment (amchandn) [reply specification/recoveryservices/resource-manager/readme.md:46]: Thanks, I've added suppression for that

**Comment by tadelesh** [comment]
Approved for Go breaking, external model removal could be ignored.

**Comment by xiaoxuqi-ms** [comment]
@amchandn , ping me once all apis have been deployed to prod

**Comment by amchandn** [comment]
> @amchandn , ping me once all apis have been deployed to prod

Thanks @xiaoxuqi-ms , we'll let you know our APIs are deployed.
Also, due to unavoidable circumstances and technical conflicts, we had to rename our API version to 2022-09-30-preview from 2022-10-02-preview

**Comment by xiaoxuqi-ms** [comment]
For model validation, certificate should be base64 encoded.

**Comment by amchandn** [comment]
> For model validation, certificate should be base64 encoded.

Thanks, done

**Review by xiaoxuqi-ms — APPROVED** [review]

**Comment by xiaoxuqi-ms** [comment]
@amchandn there are still errors of api tests, since it is staging, the errors will not block this pr but please take a look at the errors and make sure the service aligns with the swagger definition.

**Pull request merged**